File: TOMIMSV4/Entity/Transacciones/Transaccion_Ubicacion_HH/Transaccion_Ubicacion_HH_Stock/clsBeTrans_ubic_hh_stock.vb
Public Class clsBeTrans_ubic_hh_stock
    Implements ICloneable

    Public Property IdStockTransUbicHHDet() As Integer = 0
    Public Property IdTareaUbicacionEnc() As Integer = 0
    Public Property IdTareaUbicacionDet() As Integer = 0
    Public Property IdStock() As Integer = 0
    Public Property IdPropietarioBodega() As Integer = 0
    Public Property IdProductoBodega() As Integer = 0
    Public Property IdProductoEstado() As Integer = 0
    Public Property IdPresentacion() As Integer = 0
    Public Property IdUnidadMedida() As Integer = 0
    Public Property IdUbicacion() As Integer = 0
    Public Property IdUbicacion_anterior() As Integer = 0
    Public Property IdRecepcionEnc() As Integer = 0
    Public Property IdRecepcionDet() As Integer = 0
    Public Property IdPedidoEnc() As Integer = 0
    Public Property IdPickingEnc() As Integer = 0
    Public Property IdDespachoEnc() As Integer = 0
    Public Property Lote() As String = ""
    Public Property Lic_plate() As String = ""
    Public Property Serial() As String = ""
    Public Property Cantidad() As Double = 0.0
    Public Property Fecha_ingreso() As Date = Date.Now
    Public Property Fecha_vence() As Date = Date.Now
    Public Property Uds_lic_plate() As Double = 0
    Public Property No_bulto() As Integer = 0
    Public Property Fecha_manufactura() As Date = Date.Now
    Public Property añada() As Integer = 0
    Public Property User_agr() As String = ""
    Public Property Fec_agr() As Date = Date.Now
    Public Property User_mod() As String = ""
    Public Property Fec_mod() As Date = Date.Now
    Public Property Activo() As Boolean = False
    Public Property Peso() As Double = 0.0
    Public Property Temperatura() As Double = 0.0
    Public Property Fecha_mov_hist() As Date = Date.Now
    Public Property Atributo_variante_1() As String = ""
    Public Property IdProductoTallaColor() As Integer = 0
    Public Property Talla() As String = String.Empty
    Public Property Color() As String = String.Empty

    Sub New()
    End Sub

    Sub New(ByRef IdStockTransUbicHHDet As Integer, ByVal IdTareaUbicacionEnc As Integer, ByVal IdTareaUbicacionDet As Integer, ByVal IdStock As Integer, ByVal IdPropietarioBodega As Integer, ByVal IdProductoBodega As Integer, ByVal IdProductoEstado As Integer, ByVal IdPresentacion As Integer, ByVal IdUnidadMedida As Integer, ByVal IdUbicacion As Integer, ByVal IdUbicacion_anterior As Integer, ByVal IdRecepcionEnc As Integer, ByVal IdRecepcionDet As Integer, ByVal IdPedidoEnc As Integer, ByVal IdPickingEnc As Integer, ByVal IdDespachoEnc As Integer, ByVal lote As String, ByVal lic_plate As String, ByVal serial As String, ByVal cantidad As Double, ByVal fecha_ingreso As Date, ByVal fecha_vence As Date, ByVal uds_lic_plate As String, ByVal no_bulto As String, ByVal fecha_manufactura As Date, ByVal añada As Integer, ByVal user_agr As String, ByVal fec_agr As Date, ByVal user_mod As String, ByVal fec_mod As Date, ByVal activo As Boolean, ByVal peso As Double, ByVal temperatura As Double, ByVal fecha_mov_hist As Date, ByVal atributo_variante_1 As String)
        Me.IdStockTransUbicHHDet = IdStockTransUbicHHDet
        Me.IdTareaUbicacionEnc = IdTareaUbicacionEnc
        Me.IdTareaUbicacionDet = IdTareaUbicacionDet
        Me.IdStock = IdStock
        Me.IdPropietarioBodega = IdPropietarioBodega
        Me.IdProductoBodega = IdProductoBodega
        Me.IdProductoEstado = IdProductoEstado
        Me.IdPresentacion = IdPresentacion
        Me.IdUnidadMedida = IdUnidadMedida
        Me.IdUbicacion = IdUbicacion
        Me.IdUbicacion_anterior = IdUbicacion_anterior
        Me.IdRecepcionEnc = IdRecepcionEnc
        Me.IdRecepcionDet = IdRecepcionDet
        Me.IdPedidoEnc = IdPedidoEnc
        Me.IdPickingEnc = IdPickingEnc
        Me.IdDespachoEnc = IdDespachoEnc
        Me.Lote = lote
        Me.Lic_plate = lic_plate
        Me.Serial = serial
        Me.Cantidad = cantidad
        Me.Fecha_ingreso = fecha_ingreso
        Me.Fecha_vence = fecha_vence
        Me.Uds_lic_plate = uds_lic_plate
        Me.No_bulto = no_bulto
        Me.Fecha_manufactura = fecha_manufactura
        Me.añada = añada
        Me.User_agr = user_agr
        Me.Fec_agr = fec_agr
        Me.User_mod = user_mod
        Me.Fec_mod = fec_mod
        Me.Activo = activo
        Me.Peso = peso
        Me.Temperatura = temperatura
        Me.Fecha_mov_hist = fecha_mov_hist
        Me.Atributo_variante_1 = atributo_variante_1
    End Sub

    Public Function Clone() As Object Implements System.ICloneable.Clone
        Return MyBase.MemberwiseClone()
    End Function

End Class

File: WMS.EntityCore/Cambio_Ubicacion/clsBeTrans_ubic_hh_stock.cs
namespace WMS.EntityCore.Cambio_Ubicacion
{
    public class clsBeTrans_ubic_hh_stock : ICloneable
    {
        public int IdStockTransUbicHHDet { get; set; } = 0;
        public int IdTareaUbicacionEnc { get; set; } = 0;
        public int IdTareaUbicacionDet { get; set; } = 0;
        public int IdStock { get; set; } = 0;
        public int IdPropietarioBodega { get; set; } = 0;
        public int IdProductoBodega { get; set; } = 0;
        public int IdProductoEstado { get; set; } = 0;
        public int IdPresentacion { get; set; } = 0;
        public int IdUnidadMedida { get; set; } = 0;
        public int IdUbicacion { get; set; } = 0;
        public int IdUbicacion_anterior { get; set; } = 0;
        public int IdRecepcionEnc { get; set; } = 0;
        public int IdRecepcionDet { get; set; } = 0;
        public int IdPedidoEnc { get; set; } = 0;
        public int IdPickingEnc { get; set; } = 0;
        public int IdDespachoEnc { get; set; } = 0;
        public string Lote { get; set; } = "";
        public string Lic_plate { get; set; } = "";
        public string Serial { get; set; } = "";
        public double Cantidad { get; set; } = 0.0;
        public DateOnly Fecha_ingreso { get; set; } = new DateOnly(1900,1,1);
        public DateOnly Fecha_vence { get; set; } = new DateOnly(1900, 1, 1);
        public double Uds_lic_plate { get; set; } = 0;
        public int No_bulto { get; set; } = 0;
        public DateOnly Fecha_manufactura { get; set; } = new DateOnly(1900, 1, 1);
        public int añada { get; set; } = 0;
        public string User_agr { get; set; } = "";
        public DateTime Fec_agr { get; set; } = DateTime.Now;
        public string User_mod { get; set; } = "";
        public DateTime Fec_mod { get; set; } = DateTime.Now;
        public bool Activo { get; set; } = false;
        public double Peso { get; set; } = 0.0;
        public double Temperatura { get; set; } = 0.0;
        public DateTime Fecha_mov_hist { get; set; } = DateTime.Now;
        public string Atributo_variante_1 { get; set; } = "";
        public int IdProductoTallaColor { get; set; } = 0;
        public string Talla { get; set; } = string.Empty;
        public string Color { get; set; } = string.Empty;

        public clsBeTrans_ubic_hh_stock()
        {
        }

        public clsBeTrans_ubic_hh_stock(
            int IdStockTransUbicHHDet,
            int IdTareaUbicacionEnc,
            int IdTareaUbicacionDet,
            int IdStock,
            int IdPropietarioBodega,
            int IdProductoBodega,
            int IdProductoEstado,
            int IdPresentacion,
            int IdUnidadMedida,
            int IdUbicacion,
            int IdUbicacion_anterior,
            int IdRecepcionEnc,
            int IdRecepcionDet,
            int IdPedidoEnc,
            int IdPickingEnc,
            int IdDespachoEnc,
            string lote,
            string lic_plate,
            string serial,
            double cantidad,
            DateOnly fecha_ingreso,
            DateOnly fecha_vence,
            double uds_lic_plate,
            int no_bulto,
            DateOnly fecha_manufactura,
            int añada,
            string user_agr,
            DateTime fec_agr,
            string user_mod,
            DateTime fec_mod,
            bool activo,
            double peso,
            double temperatura,
            DateTime fecha_mov_hist,
            string atributo_variante_1)
        {
            this.IdStockTransUbicHHDet = IdStockTransUbicHHDet;
            this.IdTareaUbicacionEnc = IdTareaUbicacionEnc;
            this.IdTareaUbicacionDet = IdTareaUbicacionDet;
            this.IdStock = IdStock;
            this.IdPropietarioBodega = IdPropietarioBodega;
            this.IdProductoBodega = IdProductoBodega;
            this.IdProductoEstado = IdProductoEstado;
            this.IdPresentacion = IdPresentacion;
            this.IdUnidadMedida = IdUnidadMedida;
            this.IdUbicacion = IdUbicacion;
            this.IdUbicacion_anterior = IdUbicacion_anterior;
            this.IdRecepcionEnc = IdRecepcionEnc;
            this.IdRecepcionDet = IdRecepcionDet;
            this.IdPedidoEnc = IdPedidoEnc;
            this.IdPickingEnc = IdPickingEnc;
            this.IdDespachoEnc = IdDespachoEnc;
            this.Lote = lote;
            this.Lic_plate = lic_plate;
            this.Serial = serial;
            this.Cantidad = cantidad;
            this.Fecha_ingreso = fecha_ingreso;
            this.Fecha_vence = fecha_vence;
            this.Uds_lic_plate = uds_lic_plate;
            this.No_bulto = no_bulto;
            this.Fecha_manufactura = fecha_manufactura;
            this.añada = añada;
            this.User_agr = user_agr;
            this.Fec_agr = fec_agr;
            this.User_mod = user_mod;
            this.Fec_mod = fec_mod;
            this.Activo = activo;
            this.Peso = peso;
            this.Temperatura = temperatura;
            this.Fecha_mov_hist = fecha_mov_hist;
            this.Atributo_variante_1 = atributo_variante_1;
        }

        public object Clone()
        {
            return this.MemberwiseClone();
        }
    }
}